cdc/sinkv2/eventsink/mq/mq_sink.go

```diff
@@ -14,24 +14,122 @@
 package mq
 
 import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/contextutil"
 	"github.com/pingcap/tiflow/cdc/model"
+	mqv1 "github.com/pingcap/tiflow/cdc/sink/mq"
+	"github.com/pingcap/tiflow/cdc/sink/mq/codec"
+	"github.com/pingcap/tiflow/cdc/sink/mq/dispatcher"
+	"github.com/pingcap/tiflow/cdc/sink/mq/manager"
 	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink"
+	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq/producer"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/util"
+	"go.uber.org/zap"
 )
 
+// newProducerImpl specifies the build method for the producer.
+var newProducerImpl producer.NewProducerFunc = producer.NewMockProducer
+
 // Assert EventSink[E event.TableEvent] implementation
-var _ eventsink.EventSink[*model.RowChangedEvent] = (*Sink)(nil)
+var _ eventsink.EventSink[*model.RowChangedEvent] = (*sink)(nil)
+
+// sink is the mq sink.
+// It will send the events to the MQ system.
+type sink struct {
+	// id indicates this sink belongs to which processor(changefeed).
+	id model.ChangeFeedID
+	// role indicates this sink used for what.
+	role util.Role
```

Comment:
what is the purpose of the  field "role" ?

Reply:
It useful when we debugging.